src/core/lib/iomgr/resource_quota.cc

```diff
@@ -940,6 +988,12 @@ void grpc_resource_user_slice_allocator_init(
 void grpc_resource_user_alloc_slices(
     grpc_resource_user_slice_allocator* slice_allocator, size_t length,
     size_t count, grpc_slice_buffer* dest) {
+  if (gpr_atm_no_barrier_load(&slice_allocator->resource_user->shutdown)) {
```

Comment:
i'm guessing this is a new optimization kind of thing.

Reply:
It's to fix some test failure.